pkg/oc/cli/debug/debug.go

```diff
@@ -725,6 +740,66 @@ func containerNames(pod *corev1.Pod) []string {
 
 func (o *DebugOptions) approximatePodTemplateForObject(object runtime.Object) (*corev1.PodTemplateSpec, error) {
 	switch t := object.(type) {
+	case *corev1.Node:
+		if len(o.NodeName) > 0 {
+			return nil, fmt.Errorf("you may not set --node-name when debugging a node")
+		}
+		if o.AsNonRoot || o.AsUser > 0 {
+			// TODO: allow --as-root=false to skip all the namespaces except network
+			return nil, fmt.Errorf("can't debug nodes without running as the root user")
+		}
+		image := "registry.access.redhat.com/rhel7/rhel-tools"
```

Comment:
`image := o.Image` and then check that in the if below, otherwise you never allow specifying this.

Reply:
restructured it